W-13600668:  Seamless setup mule management (#56)
mulesoft/docs-code-builder: modules/ROOT/pages/manage-mule-runtime.adoc

* *Mule Home*: The full path that points to the root files of the Mule Runtime instance. +
Default value: `/Users/<user-name>/muleRuntimes/<mule-runtime-installation>`.

[WARNING]
--
Changing the *Mule Runtimes Directory* and *Mule Home* properties causes unexpected errors.

Modify these values only if you encounter any of the errors detailed in this article.
--

== Troubleshooting Mule Runtime Errors

If you encounter runtime errors when developing an integration or implementing an API, troubleshoot them as follows:

=== Mule Runtime Is Not Installed, You Must Configure a Mule Runtime to Enable This Feature

This error happens when you attempt to use a feature that requires a working Mule Runtime engine configured. For example, attempting to use auto-complete, or run a debug session. It's possible that one or both of the *Mule Home* and *Mule Runtimes Directory* are misconfigured, or empty. To troubleshoot this:

. Navigate to your Anypoint Code Builder view, and select the *Open ACB Settings* icon.
+
image::open-settings.png[]
. Locate the *Mule > Runtime: Mule Home* and *Mule > Runtime: Mule Runtimes Directory* properties:
+
image::mule-runtimes-settings.png[]

> . Ensure that both fields are not empty, and that the configurations point to an actual ZIP file containing Mule runtime.

Reviewer: **ZIP file** should be replaced by **folder**

Reviewer 2: @greobasco Is it the full path, including the folder that contains Mule, or is it to a named folder?

Reviewer: Yes, the full path. Examples can be viewed in the picture:
https://dev-docs-internal.kqa.msap.io/beta-code-builder/anypoint-code-builder/_images/mule-runtimes-settings.png

The **Mule Runtimes Directory** contains the full path to the folder that contains the Mule runtimes:
i.e: `/home/mule-dx/muleRuntimes`

The **Mule Home** contains the full path to the specific runtime: 
i.e: `/home/mule-dx/muleRuntimes/mule-enterprise-standalone-4.4.0-20230417/`

> . Ensure that both fields are not empty, and that the configurations point to an actual ZIP file containing Mule runtime.

Reviewer 2: ```suggestion
. Provide the full path to the folder that contains Mule runtime. This field must not be empty. 
```

=== No Mule runtime found. To perform this action, you must have Mule runtime configured.

This warning happens when you open Anypoint Code Builder without a Mule Runtime engine configured. It's possible that one or both of the *Mule Home* and *Mule Runtimes Directory* are misconfigured, or empty. To troubleshoot this:

. Navigate to your Anypoint Code Builder view, and select the *Open ACB Settings* icon.
+
image::open-settings.png[]
. Locate the *Mule > Runtime: Mule Home* and *Mule > Runtime: Mule Runtimes Directory* properties:
+
image::mule-runtimes-settings.png[]

> . Ensure that both fields are not empty, and that the configurations point to an actual ZIP file containing Mule runtime.

Reviewer: **ZIP file** should be replaced by **folder**

> . Ensure that both fields are not empty, and that the configurations point to an actual ZIP file containing Mule runtime.

Reviewer 2: ```suggestion
. Provide the full path to the folder that contains Mule runtime. This field must not be empty. 
```

= Mule Runtime Not Found or Not Installed Errors

Anypoint Code Builder bundles a version 4 instance of Mule runtime engine (Mule) to run your integrations and implementations.  

Do not change the default locations of Mule are unless you receive an error that Mule is not found or installed. If such errors occur, 

> the following properties enable you to change the location:

Reviewer 2: ```suggestion
the following properties enable you to change the path to Mule:
```

> * *Mule Runtimes Directory*: The default location of the Mule runtime engine installation. +

Reviewer 2: ```suggestion
* *Mule Runtimes Directory*: The default location of the Mule installation. +
```

Default value: `/Users/<user-name>/muleRuntimes/`.

> * *Mule Home*: The full path that points to the root files of the Mule Runtime instance. +

Reviewer 2: ```suggestion
* *Mule Home*: The full path that points to the root files of the Mule instance. +
```

Default value: `/Users/<user-name>/muleRuntimes/<mule-runtime-installation>`.

== Mule Runtime Is Not Installed, You Must Configure a Mule Runtime to Enable This Feature

> This error occurs if *Mule Home* and *Mule Runtimes Directory* are misconfigured or empty and feature requires a running Mule instance. 

Reviewer 2: ```suggestion
This error occurs if *Mule Home* and *Mule Runtimes Directory* are misconfigured or empty and a feature requires a running Mule instance. 
```
